cmd/thv-operator/controllers/mcpserver_controller.go

```diff
@@ -581,22 +615,18 @@ func (r *MCPServerReconciler) deploymentForMCPServer(m *mcpv1alpha1.MCPServer) *
 		}
 	}
 
-	// Prepare ProxyRunner's pod and container security context
-	proxyRunnerPodSecurityContext := &corev1.PodSecurityContext{
-		RunAsNonRoot: ptr.To(true),
-		RunAsUser:    ptr.To(int64(1000)),
-		RunAsGroup:   ptr.To(int64(1000)),
-		FSGroup:      ptr.To(int64(1000)),
+	// Detect platform and prepare ProxyRunner's pod and container security context
+	platform, err := r.detectPlatform(ctx)
```

Comment:
maybe, we don't need both the `platform` local variable and the `detectedPlatform` field in `r`, WDYT?

Reply:
Addressed as per the updated [mcpserver_controller.go#L619](https://github.com/stacklok/toolhive/blob/d4e8cce60860380cb39508a96607ea61f342a14e/cmd/thv-operator/controllers/mcpserver_controller.go#L619)